src/Symfony/Component/Validator/Constraints/Base64Validator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class Base64Validator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Base64) {
+            throw new UnexpectedTypeException($constraint, Base64::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!\is_string($value) && !$value instanceof \Stringable) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        if ($constraint->allowDataUri) {
+            $value = preg_replace('/^data:.+;base64,/', '', $value);
+        }
+
+        if (false === base64_decode($value, true)) {
```

Comment:
Ensure `false` is strictly returned because an empty string is a valid base64 string

Reply:
This way of checking consumes memory for no good reasons because it generates the decoded payload and trashes is right away.
A regexp would be better.
Also, there are two styles of base64 encoding (url-safe and not-url-safe), so we need a way to tell which one we want.

Reply:
I updated the check to use a regex to check base64 validity.

Also, the `urlEncoded` option has been added. Is that what you had in mind?